lib/train/extras/command_wrapper.rb

```diff
@@ -162,7 +162,13 @@ class CommandWrapper
     include_options WindowsCommand
 
     def self.load(transport, options)
-      if transport.os.unix?
+      if transport.class == Train::Transports::SSH::Connection && !transport.connected?
```

Comment:
I'm pretty new so I'm not sure my analysis of train itself is all that accurate... That said:

It looks to me like it is `transport.os.unix?` itself that is the method call that causes an actual connection to be established, so having this code before that call is ensuring that `transport.connected?` is false.

Reply:
> Thank you for this PR!
> 
> I'm not sure this is the right place to put this check (but I'm new, so...). I've also made some other recommendation to make the ruby more idiomatic. Finally, I think this needs tests to show how it works in both the positive and exceptional cases.

I was also not sure whether this the right place to do this. So I was debugging into the code more and also wanted some input so that I get right direction. I have removed this now and fixed in the os_common.rb file where the run_command is executing when we call `transport.os.unix?` in command_wrapper.rb. So as this is the fix for ssh connection I am checking the class of the connection. Second as the command execution with root user for ssh connection results in standard output `Please login as the user "azure" rather than the user "root"` I am doing string match. This looks to be a right fix as it raises appropriate error to user on root level. But Ia m open for suggestions. I am trying to understand the existing test suite so that I can add code coverage to this.